Gems/LyShine/Code/Source/UiRenderer.cpp

```diff
@@ -492,29 +451,52 @@ void UiRenderer::DebugDisplayTextureData(int recordingOption)
         {
             CDraw2d::TextOptions textOptions = draw2d->GetDefaultTextOptions();
             textOptions.color = color;
-            AZ::Vector2 textSize = draw2d->GetTextSize(buffer, 16, &textOptions);
+            AZ::Vector2 textSize = draw2d->GetTextSize(buffer, fontSize, &textOptions);
             AZ::Vector2 rectTopLeft = AZ::Vector2(xOffset - 2, yOffset);
             AZ::Vector2 rectSize = AZ::Vector2(textSize.GetX() + 4, lineSpacing);
             draw2d->DrawImage(blackTexture, rectTopLeft, rectSize, backgroundRectOpacity);
-            draw2d->DrawText(buffer, AZ::Vector2(xOffset, yOffset), 16, textOpacity, &textOptions);
+            draw2d->DrawText(buffer, AZ::Vector2(xOffset, yOffset), fontSize, textOpacity, &textOptions);
             yOffset += lineSpacing;
         };
 
-        int numTexturesUsedInFrame = m_texturesUsedInFrame.size();
+        size_t numTexturesUsedInFrame = m_texturesUsedInFrame.size();
         char buffer[200];
-        sprintf_s(buffer, "There are %d unique UI textures rendered in this frame, the total texture area is %d (%d x %d), total data size is %d (%.2f MB)",
+        sprintf_s(buffer, "There are %zu unique UI textures rendered in this frame, the total texture area is %d (%d x %d), total data size is %d (%.2f MB)",
             numTexturesUsedInFrame, totalArea, xDim, yDim, totalDataSize, totalDataSizeMB);
         WriteLine(buffer, white);
-        sprintf_s(buffer, "Dimensions   Data Size   Format Texture name");
+        sprintf_s(buffer, "Dimensions   Data Size    Format Texture name");
         WriteLine(buffer, blue);
 
-        for (ITexture* texture : textures)
+        for (auto texture : textures)
         {
-            sprintf_s(buffer, "%4d x %4d, %9d %8s %s",
-                texture->GetWidth(), texture->GetHeight(), texture->GetDataSize(), texture->GetFormatName(), texture->GetName());
+            AZ::Data::Instance<AZ::RPI::Image> image = texture.first;
+            const AZ::RHI::ImageDescriptor& imageDescriptor = image->GetRHIImage()->GetDescriptor();
+            AZ::RHI::Size size = imageDescriptor.m_size;
+            uint32_t width = imageDescriptor.m_size.m_width;
+            uint32_t height = imageDescriptor.m_size.m_height;
+            uint32_t dataSize = texture.second;
+
+            // Check if the image has been assigned a name (ex. attachment image or cpu generated image)
+            AZStd::string textureName = image->GetRHIImage()->GetName().GetStringView();
```

Comment:
nit: this is an extra string copy, probably not a big deal here but worth keeping an eye on

Reply:
Yeah since GetAssetPathById returns an AZStd::string (and this is for debug use), I went with the simpler logic, but I've now updated it to what I believe is a little more optimized. Let me know if you meant something else.